vars/provisionNodes.groovy

```diff
@@ -57,14 +57,16 @@ def call(Map config = [:]) {
       return node_cnt
   }
 
-  def distro = config.get('distro', 'el7')
+  def distro = config.get('distro', 'centos7')
   def inst_rpms = config.get('inst_rpms', '')
   def inst_repos = config.get('inst_repos','')
 
   def repository_g = ''
   def repository_l = ''
   def distro_type = 'el'
-  if (distro.startsWith("sles") || distro.startsWith("leap") ||
+  if (distro == 'el7') {
+    distro = 'centos7'  //Backwards compatibility
+  } else if (distro.startsWith("sles") || distro.startsWith("leap") ||
       distro.startsWith("opensuse")) {
       distro_type = 'suse'
```

Comment:
It is making it consistent with the majority of existing use in the Jenkinsfile and pipeline-liib.
We are using CentOS or centos for the stage names and as filename components.  Making everything in the Jenkinsfile and pipeline-lib parameters being "centos" instead of "el" makes it consistent.

If we were to change to standardize on "el" instead of "centos" then we would have to change the names that github is expecting status results for in PRs.

For suse family the 'suse' is used internally to the provisionNodes.   The 'sles' name would only be used in the Jenkinsfile if we actually get setup to do provisioning for SLES.   With out that, everything in the Jenkinsfile refers to the operating system as "leap" to not confuse it with "opensuse Tumbleweed" and past Suse naming conventions.

The provisionNodes needs to be able to handle both 'el7' and 'centos7' the same way.

Reply:
> It is making it consistent with the majority of existing use in the Jenkinsfile and pipeline-liib.

Yeah, I'm not saying that existing use is at all consistent either though.  Indeed, we do seem to be currently using a generic name for RHEL/CentOS, and specific per-distro names for Leap/SLES.  This is just what happened organicaly.

I don't disagree necessarily with the idea of multiple distro generic names vs. individual distro names, just that we be consistent.  And we have not been and don't seem to be heading in a direction of consistency with these changes.

> We are using CentOS or centos for the stage names and as filename components. Making everything in the Jenkinsfile and pipeline-lib parameters being "centos" instead of "el" makes it consistent.

But makes it the opposite of the change you are proposing to group SLES and Leap into a single _distro_type_ of `suse`.

> If we were to change to standardize on "el" instead of "centos" then we would have to change the names that github is expecting status results for in PRs.

Which is not impossible, by any means.

> For suse family the 'suse' is used internally to the provisionNodes. The 'sles' name would only be used in the Jenkinsfile if we actually get setup to do provisioning for SLES.

Don't disagree with this point.  But to be consistent then we should be using a family name of _el_ and only use `rhel` in `Jenkinsfile` for actual RHEL nodes.  Not that I think using bona fide RHEL nor SLES are terribly important.  I think we can rely on their general compatibility with each other for our purposes.  We don't do anything with the RHEL (and SLES, AFAIK) differences to CentOS.

> With out that, everything in the Jenkinsfile refers to the operating system as "leap" to not confuse it with "opensuse Tumbleweed" and past Suse naming conventions.

Again, not disagreeing with the change of the uses of _leap_ to _suse_, but we should be changing _centos_ to _el_ to be consistent.

Reply:
I was not changing a use of leap to suse.  I was using distro_type of either "suse" or "el" to determine the different handling of the families.

I am using 'centos' when we are specifically using CentOS.  As is most of the Jenkinsfile portion of the DAOS project.

It is a minor change to DAOS to change the few Jenkinsfiles references to "el7" where it specifically is only used for "centos7" to make them consistent for a Matrix implementation.
I do not see anything that is noticeable external to the Jenkinsfile.

It is a much more significant change to DAOS to change all the "centos7" or "CentOS"  references to "el7" references.  Renaming the CentOS dockerfiles, changing what context names that github is expecting to complete, etc.  There would be a period of time that PRs would fail validation, and we would need more new to old name translations in the pipeline lib to handle building older releases.   The visible names on the Jenkins stages would change.

This routine, will accept either el or centos and do the same thing.  The default of centos7 or el7 really depends on which of the above action is going to be chosen for the daos project.

Reply:
Maybe what is sticking out like a sore thumb here is the inconsistent/intermixed use of `distro` and `distro_type` in this file to determine the branches to take such as:
https://github.com/daos-stack/pipeline-lib/blob/d38fa3177142febee0922e28eecbf7936b512ff1/vars/provisionNodes.groovy#L181
and then:
https://github.com/daos-stack/pipeline-lib/blob/d38fa3177142febee0922e28eecbf7936b512ff1/vars/provisionNodes.groovy#L269